### PR TITLE
Flagrant/sponsors bug fixes

### PR DESCRIPTION
**Pull request by codyjamesbrooks**
Reason for Change
=================
I was recording a loom video to go over the 'adding sponsors to an Event' in the CFP. And while I was recording the video there were some things that really stood out as feeling as though they really needed to be improved. 

Some of the things are small. A little bit of CSS over here, a data validation sprinkled in over there. 
Some of the things were actual oversights, that would have been flushed out with more testing and use. 

The commit message has a detailed accounting of everything that the PR seeks to accomplish, And I have included tests on all of the behavior that I noticed that I wanted to ensure was correct. 
 
Changes
=======
Here is a list of all of the Sponsor related updates
- adding padding, and max-width to sponsors containers on the sponsors page
- properly style the page title to the sponsors page
- order sponsors on index
- resize sponsor primary logo
- add sponsor star the the data that is sent back from the .js time slot form
- properly scope the query to pull an websites events sponsors on the sponsors show page
- require sponsors have a name, tier, url, description, and lastly a primary_logo
- remove 'other_title' from the sponsor model
- modify sponsors index to indicate if the sponsor has a footer_logo or a banner_ad present

Here are the updates that are not sponsor related, and are more general application cleanup
- fixed up some CSS class duplication
- allow tags on programs page to wrap rather than overflow their container

### CODE REVIEW

**Comment by adarsh** [comment]
bump @jonsgreen for review

File: spec/features/website/sponsor_spec.rb [hunk 1]
```diff
@@ -60,5 +60,16 @@
       expect(page).to_not have_content(sponsor.offer_text)
       expect(page).to_not have_content(sponsor.offer_headline)
     end
+
+    it "Sponsor page doesn't disply sponsors from a different event", js: true do
```
Comment (jonsgreen) [inline spec/features/website/sponsor_spec.rb:64]: disply/display

**Review by jonsgreen — APPROVED** [review]
@codyjamesbrooks this looks great. Thanks for making sure everything looks good sponsors!

File: spec/features/website/sponsor_spec.rb (hunk 1 above)
Comment (codyjamesbrooks) [reply spec/features/website/sponsor_spec.rb:64]: 👍

**Pull request merged**